uccb/src/bin/validator-oracle.rs

```diff
@@ -0,0 +1,273 @@
+use std::{path::PathBuf, str::FromStr};
+
+use alloy::{
+    contract::{ContractInstance, DynCallBuilder, Interface},
+    dyn_abi::DynSolValue,
+    eips::{eip2930::AccessList, BlockNumberOrTag},
+    json_abi::JsonAbi,
+    primitives::Address,
+    providers::Provider,
+    pubsub::PubSubFrontend,
+    rpc::types::Filter,
+    signers::local::PrivateKeySigner,
+};
+use anyhow::Result;
+use clap::Parser;
+use futures_util::stream::StreamExt;
+use tokio::sync::watch;
+use tracing::{debug, error, info};
+use tracing_subscriber::EnvFilter;
+use uccb::{
+    cfg::{ChainConfig, Config},
+    client::ChainClient,
+};
+use zilliqa::{contracts, crypto::SecretKey, state::contract_addr};
+
+const VALIDATOR_MANAGER_ABI_JSON: &str =
+    include_str!["../../contracts/out/ValidatorManager.sol/ValidatorManager.json"];
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(value_parser = SecretKey::from_hex)]
+    secret_key: SecretKey,
+    #[clap(long, short, default_value = "config.toml")]
+    config_file: PathBuf,
+}
+
+impl uccb::Args for Args {
```

Comment:
This is used in another (future) PR. Needed so we can extract what we need from both the args in the validator-oracle binary and the zilliqa binary.

Reply:
I still don't understand - Why do two different binaries need to share a trait for accessing their command line arguments? If they both take a secret key, then I think we should just duplicate that field in their respective structs. The trait doesn't achieve anything, since we aren't ever abstracting over that trait (we don't have any functions which take `T: uccb::Args`).

Reply:
Please don't resolve this conversation without discussing or making any changes.

Reply:
Look at the next PR. There's quite a few changes on top of this one and hopefully you'll see why.